api/app/Policies/UserPolicy.php

```diff
@@ -89,50 +89,49 @@ public function updateRoles(User $user, $args)
     {
         $targetUserId = isset($args['id']) ? $args['id'] : null;
         $attachRoles = isset($args['roleAssignmentsInput']) && isset($args['roleAssignmentsInput']['attach']) ?
-            $args['roleAssignmentsInput']['attach'] : null;
+            $args['roleAssignmentsInput']['attach'] : [];
         $detachRoles = isset($args['roleAssignmentsInput']) && isset($args['roleAssignmentsInput']['detach']) ?
-            $args['roleAssignmentsInput']['detach'] : null;
+            $args['roleAssignmentsInput']['detach'] : [];
 
         if (is_null($targetUserId)) {
             return false;
         }
 
-        if (! empty($attachRoles)) {
+        if (empty($attachRoles) && empty($detachRoles)) {
+            return true;
+        }
 
-            foreach ($attachRoles as $roleInput) {
+        $canAttachRoles = false;
+        $canDetachRoles = false;
 
-                // loop through each element and check
-                if (isset($roleInput['teamId'])) {
-                    if (! $this->teamAbleToCheck($user, $roleInput['roleId'], $roleInput['teamId'])) {
-                        return false;
-                    }
-                } else {
-                    if (! $this->individualAbleToCheck($user, $roleInput['roleId'])) {
-                        return false;
-                    }
-                }
+        $canAttachRoles = collect($attachRoles)->every(function($roleInput) use ($user) {
+            // loop through each element and check
+            if (isset($roleInput['teamId'])) {
+                return $this->teamAbleToCheck($user, $roleInput['roleId'], $roleInput['teamId']);
+            } else {
+                return $this->individualAbleToCheck($user, $roleInput['roleId']);
             }
-        }
 
-        if (! empty($detachRoles)) {
+            return true;
```

Comment:
Falling through to "true" is what we're trying to get rid of.  :grin:

Reply:
:upside_down_face: Removed in [17b2a82](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11716/commits/17b2a82d07fa6e5d12141d66e2f9b81e1cea83d0)